app/Http/Controllers/Api/ContestsController.php

```diff
@@ -19,13 +25,37 @@ class ContestsController extends ApiController
     public function __construct()
     {
         $this->transformer = new ContestTransformer;
+        $this->contestAndWaitingTransformer = new ContestAndWaitingTransformer;
     }
 
     /**
-     * Get a collection of waiting rooms.
+     * Get a collection of contests
      */
     public function index()
     {
         return 'Hello! We need to wait for the Contest CRUD to continue...';
     }
+
+    /**
+     * Get a specific contest based on run id.
+     *
+     * @param int $run_nid
+     * @return \Illuminate\Http\Response
+     */
+    public function getByRunId($run_nid)
+    {
+        // Get contest by run nid
+        $contest = Contest::where('campaign_run_id', $run_nid)->firstOrFail();
+
+        // Get waiting room for this contest
+        $waitingRoom = WaitingRoom::where('contest_id', $contest->id)->firstOrFail();
```

Comment:
I think this could be done with a relation on the model, like `$contest->waitingRoom`.